app/utils/report/test.py

```diff
@@ -56,7 +56,118 @@
 ]
 
 
+def _get_case(list_cases, name):
+    """Look for a test case with a given name
+
+    :param list_case: List of test cases from  test group
+    :type list_case: list
+    :param name: Name of a test case
+    :type name: string
+    """
+
+    for d in list_cases:
+        if d['name'] == name:
+            return d
+
+
+def create_regressions_data(test_docs, test_data):
+    """Create the regressions data for the email report.
+
+    Will create a dictionary with all the regressions
+
+    :param test_docs: The regressions data (the list of tests).
+    :type test_docs: list
+    :param test_data: Details about the test results
+    :type test_data: dict
+    :return dict The regressions results in a dictionary.
+    """
+    regr_dict = {}
+
+    # Make sure the test reports in the regressions data structure are
+    # correctly sorted by date, so that the oldest document is the PASS one
+    # and the most recent one is the last FAIL-ed.
+    test_docs.sort(
+        cmp=lambda x, y: cmp(x[models.CREATED_KEY], y[models.CREATED_KEY]))
+
+    ndocs = len(test_docs)
+    last_good = test_docs[0]
+
+    if ndocs == 1:
+        # This should never happen
+        utils.LOG.warn("This message never should appear. There is \
+                        something wrong with the regression document!")
```

Comment:
Might be worth throwing an exception here?

Reply:
I don't know if I should leave these lines even. Every time the backend write a regression document, there will be at least two documents already.